test/unit/gateways/migs_test.rb

```diff
@@ -74,6 +76,26 @@ def test_scrub
     assert_equal @gateway.scrub(pre_scrubbed), post_scrubbed
   end
 
+  def test_purchase_passes_tx_source
+    @gateway.expects(:commit).with(has_entries(:TxSource => 'MOTO'))
```

Comment:
We could make these expectations more DRY by factoring them out into a method. That way, if additional methods need this test in the future, we can just call that expectation method. Likewise, if we need to change how the expectation works, we'll be able to do it in one place.

For example, something like:

```
def expect_gateway_commit_with_tx_source
  @gateway.expects(:commit).with(has_entries(:TxSource => @tx_source))
end
```
with ```@tx_source``` defined once in the ```setup``` method, which is also more DRY since we could change the value in a single place if we needed to.